Src/Support/Google.Apis.Auth/OAuth2/GoogleCredential.cs

```diff
@@ -83,10 +95,13 @@ internal GoogleCredential(ICredential credential)
         /// </list>
         /// </summary>
         /// <returns>A task which completes with the application default credentials.</returns>
-        public static Task<GoogleCredential> GetApplicationDefaultAsync()
-        {
-            return defaultCredentialProvider.GetDefaultCredentialAsync();
-        }
+        public static async Task<GoogleCredential> GetApplicationDefaultAsync(IEnumerable<string> scopes) =>
+            Scoped(await defaultCredentialProvider.GetDefaultCredentialAsync(), scopes);
+
+        public static Task<GoogleCredential> GetApplicationDefaultAsync() => GetApplicationDefaultAsync(Enumerable.Empty<string>());
+
+        public static Task<GoogleCredential> GetApplicationDefaultAsync(string scope1, params string[] scopes) =>
```

Comment:
The positive effect of this new overload is that it's easier to explain scopes.  The negative effect is that now the user sees the variant with scopes, and probably worries if they need to add scopes.  I don't remember writing any code that calls GetApplicationDefaultAsync() and then add scopes.  In what situations will this be necessary?

Reply:
In our higher-level client libraries we're doing that ourselves, but anyone using just the generated libraries would have to do it.

Reply:
As I understand it, if the default credentials are service credentials, then scopes will be required.
So this means, if it'll *ever* run with service credentials then scopes are required; even if much of the time it'll be running with, say, compute credentials.